miso-service/src/main/java/uk/ac/bbsrc/tgac/miso/service/impl/DefaultPoolOrderService.java

```diff
@@ -202,29 +205,96 @@ private static boolean allMatch(Collection<OrderLibraryAliquot> one, Collection<
 
   @Override
   protected void applyChanges(PoolOrder to, PoolOrder from) throws IOException {
+    // Properties not changelogged here are caught by SQL triggers
     to.setPartitions(from.getPartitions());
+
+    if (from.getParameters() == null && to.getParameters() != null) {
+      changeLogService.create(
+              to.createChangeLog("Sequencing Parameters deleted.",
+                      "parametersId",
+                      authorizationManager.getCurrentUser())
+      );
+    } else if(to.getParameters() == null && from.getParameters() != null){
+      changeLogService.create(
+              to.createChangeLog("New Sequencing Parameters:"
+                      + from.getParameters().getInstrumentModel().getAlias()
+                      + ", " + from.getParameters().getName(),
+                      "parametersId",
+                      authorizationManager.getCurrentUser())
+      );
+    } else if(to.getParameters() != null && from.getParameters() != null
+            && !to.getParameters().equals(from.getParameters())) {
+      changeLogService.create(
+              to.createChangeLog("Changed Sequencing Parameters: "
+                              + to.getParameters().getInstrumentModel().getAlias()
+                              + ", " + to.getParameters().getName()+ " to "
+                              + from.getParameters().getInstrumentModel().getAlias()
+                              + ", " + from.getParameters().getName(),
+                      "parametersId",
+                      authorizationManager.getCurrentUser())
+      );
+    }
     to.setParameters(from.getParameters());
     to.setAlias(from.getAlias());
+
+    if(!to.getPurpose().getAlias().equals(from.getPurpose().getAlias())) changeLogService.create(
+            to.createChangeLog("Changed Purpose: "
+                            + to.getPurpose().getAlias()
+                            + " to " + from.getPurpose().getAlias(),
+                    "purposeId",
+                    authorizationManager.getCurrentUser())
+    );
     to.setPurpose(from.getPurpose());
     to.setDescription(from.getDescription());
     to.setDraft(from.isDraft());
 
     // remove from TO if library isn't in FROM
     // add to TO if library isn't in TO
     // update proportion in TO if different in FROM
-    to.getOrderLibraryAliquots().removeIf(toOrderAli -> from.getOrderLibraryAliquots().stream()
-        .noneMatch(fromOrderAli -> toOrderAli.getAliquot().getId() == fromOrderAli.getAliquot().getId()));
+    Set<Long> fromIds = new HashSet<>();
+    for (OrderLibraryAliquot fromOrderAli : from.getOrderLibraryAliquots()){
+      fromIds.add(fromOrderAli.getAliquot().getId());
```

Comment:
I wondering if, for all the add/remove/change operations, there should be 1 aggregate message. That's how we do it for boxes... @callunity?

Reply:
I was debating that, and looked at boxes as well -- it looks to my eye like they do individual messages though? (I'm looking at box 1 on staging). Although aggregating would lead to fewer messages, which is a positive, I find it more understandable to scan vertically rather than horizontally...

Reply:
(And ultimately, Pool Orders will have fewer changelogs than boxes)